api/status.go

```diff
@@ -0,0 +1,209 @@
+				memberStatuses, err := client.GetStatus(ctx, c)
+				if err != nil {
+					return err
+				}
+
+				statuses = append(statuses, memberStatuses...)
+
+				return nil
+			})
+			if err != nil {
+				return response.SmartError(err)
+			}
+		}
+
+		status := types.Status{
+			Name:         sh.Name,
+			Address:      sh.Address,
+			Clusters:     map[types.ServiceType][]microTypes.ClusterMember{},
+			OSDs:         []cephTypes.Disk{},
+			CephServices: []cephTypes.Service{},
+			OVNServices:  []ovnTypes.Service{},
+		}
+
+		err := sh.RunConcurrent("", "", func(s service.Service) error {
+			var err error
+			var microClient *microClient.Client
+			var lxd lxd.InstanceServer
+			switch s.Type() {
+			case types.LXD:
+				lxd, err = s.(*service.LXDService).Client(context.Background())
+			case types.MicroCeph:
+				microClient, err = s.(*service.CephService).Client("")
+			case types.MicroOVN:
+				microClient, err = s.(*service.OVNService).Client()
+			case types.MicroCloud:
+				microClient, err = s.(*service.CloudService).Client()
+			}
+			if err != nil {
+				return err
+			}
+
+			if microClient != nil {
+				clusterMembers, err := microClient.GetClusterMembers(context.Background())
+				if err != nil && !lxdAPI.StatusErrorCheck(err, http.StatusServiceUnavailable) {
+					return err
+				}
+
+				status.Clusters[s.Type()] = clusterMembers
+
+				if s.Type() == types.MicroCeph {
+					disks, err := cephClient.GetDisks(r.Context(), microClient)
+					if err != nil {
+						return err
+					}
+
+					for _, disk := range disks {
+						if disk.Location == s.Name() {
+							status.OSDs = append(status.OSDs, disk)
+						}
+					}
+
+					services, err := cephClient.GetServices(r.Context(), microClient)
+					if err != nil {
+						return err
+					}
+
+					for _, service := range services {
+						if service.Location == s.Name() {
+							status.CephServices = append(status.CephServices, service)
+						}
+					}
+				}
+
+				if s.Type() == types.MicroOVN {
+					services, err := ovnClient.GetServices(r.Context(), microClient)
+					if err != nil {
+						return err
+					}
+
+					for _, service := range services {
+						if service.Location == s.Name() {
+							status.OVNServices = append(status.OVNServices, service)
+						}
+					}
+				}
+
+			} else if lxd != nil {
+				server, _, err := lxd.GetServer()
+				if err != nil {
+					return err
+				}
+
+				if server.Environment.ServerClustered {
+					clusterMembers, err := lxd.GetClusterMembers()
+					if err != nil {
+						return err
+					}
+
+					certs, err := lxd.GetCertificates()
+					if err != nil {
+						return err
+					}
+
+					microMembers := make([]microTypes.ClusterMember, 0, len(clusterMembers))
+					for _, member := range clusterMembers {
+						url, err := url.Parse(member.URL)
+						if err != nil {
+							return err
+						}
+
+						addrPort, err := microTypes.ParseAddrPort(util.CanonicalNetworkAddress(url.Host, 8443))
```

Comment:
Should we expect that the user has never changed the LXD port or dynamically load the right `core.https_address` from the server config? Maybe it's even contained in the `server` var above.

Aside from this, why does the address have to be populated here? Is it used later to access LXD when parsing the reponse from this endpoint?

Reply:
Actually we have a constant for each services' port which we should be using here.

The main reason address is populated here is because it's populated for the `microcluster` projects, so I wanted the returned struct from LXD to be as close as possible to those in returned information, even if we're not using it all right now.